src/main/scala/nl.knaw.dans.easy.bagindex/server/BagIndexServletComponent.scala

```diff
@@ -136,9 +136,8 @@ trait BagIndexServletComponent {
         .flatMap(uuid => databaseAccess.doTransaction(implicit c => database.getBagInfo(uuid)))
         .map(createResponse[BagInfo](bagInfo => <result>{toXml(bagInfo)}</result>)(bagInfo => "result" -> toJson(bagInfo)))
         .recoverWith {
-          case BagIdNotFoundException(_) => Success(createResponse[Unit](_ => <result/>)(_ => "result" -> JArray(List.empty))(()))
+          case BagIdNotFoundException(_) => Success(NotFound(createResponse[Unit](_ => <result/>)(_ => s"bag with id $uuidStr could not be found")(())))
```

Comment:
Now you still have an empty `<result/>` object when requesting the XML response. I don't think you need the `createResponse` call here. The following should be enough:

```suggestion
          case BagIdNotFoundException(_) => Success(NotFound(s"bag with id $uuidStr could not be found"))
```